model.py

```diff
@@ -43,12 +40,38 @@
 
 class LLourney(nn.Module):
     
-    def __init__(self):
+    def __init__(
+        self,
+        image_dim: int = 256,
+        patch_dim: int = 2,
+        context_size: int = 32,
+        vae_model_id_or_path: str = 'runwayml/stable-diffusion-v1-5',
+        llm_model_id_or_path: str = 'gpt2',
+        llm_tokenizer_model_id_or_path: str = 'gpt2',
+    ):
         super().__init__()
+        self.image_dim = image_dim
+        self.patch_dim = patch_dim
+        self.context_size = context_size
+
+        self.vae = AutoencoderKL.from_pretrained(vae_model_id_or_path, subfolder='vae')
+        self.vae_scale_factor = 2 ** (len(self.vae.config.block_out_channels) - 1)
+        self.vae_latent_channels = self.vae.config.latent_channels
+        self.latent_dim = image_dim // self.vae_scale_factor
 
-        self.vae = AutoencoderKL.from_pretrained(STABILITY_MODEL, subfolder='vae')
-        # We use the pre-trained VAE from Stable Diffusion with frozen weights
-        self.vae.requires_grad_(False)
+        self.num_patches = (self.latent_dim // self.patch_dim) ** 2
+
+        # TODO(2): Use official tokenizer and model / upgrade the tokenizer and model
+        self.llama_tokenizer = AutoTokenizer.from_pretrained(llm_tokenizer_model_id_or_path)
+        # If the tokenizer doesn't have a pad_token, set it to eos_token
+        if self.llama_tokenizer.pad_token is None:
+            print(f"Setting tokenizer pad_token to eos_token: {self.llama_tokenizer.eos_token}")
+            self.llama_tokenizer.pad_token = self.llama_tokenizer.eos_token
+        # Set padding_side explicitly (?)
+        self.llama_tokenizer.padding_side = "right"
```

Comment:
can you leave a comment here suggesting we probably want to try `padding_side = "left"` at some point?